python/tank/platform/interactive_authentication.py

```diff
@@ -14,172 +14,272 @@
 
 from getpass import getpass
 import logging
+import threading
 
-from tank.errors import TankAuthenticationError
+from tank.errors import TankAuthenticationError, TankAuthenticationDisabled
 from tank.util import authentication
 from tank.util.login import get_login_name
 from tank.util import shotgun
 
 # Configure logging
 logger = logging.getLogger("sgtk.interactive_authentication")
```

Comment:
Again, I am not totally sure about introducing loggers everywhere just yet. Maybe this is something we could add in when we address the logging epic later on?